sdk/src/main/java/com/bugsnag/android/SessionTracker.java

```diff
@@ -252,34 +243,41 @@ private void leaveBreadcrumb(String activityName, String lifecycleCallback) {
      * should be started, unless the app is within a timeout period.
      *
      * @param activityName the activity name
-     * @param inForeground whether the activity is in the foreground or not
+     * @param activityStarting whether the activity is being started or not
+     * @param nowMs The current time in ms
      */
-    void updateForegroundTracker(String activityName, boolean inForeground, long now) {
-        if (inForeground) {
-            long delta = now - lastForegroundMs;
+    void updateForegroundTracker(String activityName, boolean activityStarting, long nowMs) {
+        if (activityStarting) {
+            long noActivityRunningForMs = nowMs - activityLastStoppedAtMs.get();
+
+            //TODO:SM Race condition between isEmpty and put
```

Comment:
I didn't find a real-world case for this, even programmatically switching foreground activities rapidly.